diff --git a/node_modules/react-native-stallion/react-native-stallion.podspec b/node_modules/react-native-stallion/react-native-stallion.podspec
index c2a92cc..5c54c24 100644
--- a/node_modules/react-native-stallion/react-native-stallion.podspec
+++ b/node_modules/react-native-stallion/react-native-stallion.podspec
@@ -45,14 +45,17 @@ Pod::Spec.new do |s|
     puts "Stallion is disabled, falling back to noop version. Check your stallion.config.js file"
   end
 
-  s.dependency "React-Core"
   if isStallionEnabled then
     s.dependency "ZIPFoundation"
   end
 
+  # Use install_modules_dependencies helper to install the dependencies if React Native version >=0.71.0.
+  # See https://github.com/facebook/react-native/blob/febf6b7f33fdb4904669f99d795eba4c0f95d7bf/scripts/cocoapods/new_architecture.rb#L79.
   if respond_to?(:install_modules_dependencies, true)
     install_modules_dependencies(s)
   else
+    s.dependency "React-Core"
+
     # Don't install the dependencies when we run `pod install` in the old architecture.
     if ENV['RCT_NEW_ARCH_ENABLED'] == '1' then
       s.compiler_flags = folly_compiler_flags + " -DRCT_NEW_ARCH_ENABLED=1"
